src/Loop.php

```diff
@@ -40,13 +40,77 @@ public static function execute(callable $callback, LoopDriver $driver)
     public static function get()
     {
         if (null === self::$driver) {
-            throw new \RuntimeException('Not within the scope of an event loop driver');
+            throw new LoopNotSetException();
         }
 
         return self::$driver;
     }
 
     /**
+     * Sets the global event loop driver.
+     *
+     * Calls to the event loop that occur before this method is called will
+     * result in an error thrown, so this method should be called at the
+     * beginning of the program.
+     *
+     * Library packages should not typically call this method. When writing a
+     * concrete program, it is up to the application developer to choose what
+     * event loop implementation to use, and all libraries will use the same
+     * implementation.
+     *
+     * If an event loop is already set, you must drop it with `EventLoop::drop()`
+     * before you may replace the loop driver. It is an error to set a driver
+     * on top of an existing one.
+     *
+     * @param LoopDriver $driver Event loop instance to use as the global loop.
+     *
+     * @throws LoopAlreadySetError Thrown if the global event loop has already been set.
+     */
+    public static function set(LoopDriver $driver)
+    {
+        if (null !== self::$driver) {
+            throw new LoopAlreadySetException();
+        }
+
+        self::$driver = $driver;
+    }
+
+    /**
+     * Drops the global event loop, if any.
+     *
+     * The global event loop will be unset and discarded. This method calls
+     * `stop()` on the driver first before discarding so that it completes the
+     * current iteration safely.
+     *
+     * @return bool Indicates if an event loop was removed.
+     */
+    public static function drop()
+    {
+        if (null === self::$driver) {
+            return false;
+        }
+
+        // Stop the event loop first.
+        self::stop();
```

Comment:
This might not work. At least in Amp, `stop()` currently doesn't tick the event loop, it just makes it not entering the next loop iteration, see https://github.com/amphp/amp/blob/master/lib/NativeReactor.php#L130-L142